core/src/main/scala/org/apache/spark/resource/ResourceAllocator.scala

```diff
@@ -56,7 +56,7 @@ trait ResourceAllocator {
   def availableAddrs: Seq[String] = addressAvailabilityMap
     .flatMap { case (addr, available) =>
       (0 until available).map(_ => addr)
-    }.toSeq
+    }.toSeq.sorted
```

Comment:
we should do assignedAddrs as well to keep them consistent. There are tests using it but I didn't see